plugins/org.eclipse.n4js/src/org/eclipse/n4js/xtext/workspace/XWorkspaceConfigSnapshotProvider.java

```diff
@@ -32,12 +30,9 @@
 	 * Note that this implementation may be removed when the N4JS Eclipse product is removed.
 	 */
 	class NullXWorkspaceConfigSnapshotProvider implements XWorkspaceConfigSnapshotProvider {
-		@Inject
-		ProjectBuildOrderInfo.Provider provider;
-
 		@Override
 		public WorkspaceConfigSnapshot getWorkspaceConfigSnapshot() {
-			return new WorkspaceConfigSnapshot(null, Collections.emptyList(), provider);
+			return new ConfigSnapshotFactory().createWorkspaceConfigSnapshot((URI) null);
```

Comment:
I think class `NullXWorkspaceConfigSnapshotProvider` can use injection. If that is true, it would be better to use an injected instance of `ConfigSnapshotFactory` here.